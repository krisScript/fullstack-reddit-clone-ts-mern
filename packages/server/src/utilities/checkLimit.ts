import { ErrorREST, Errors } from '../classes/ErrorREST';
import ValidationError from '../types/ValidationError';
const checkLimit = (limit: number): void => {
  try {
    if (limit > 50 || limit < 1) {
      const errorData: ValidationError = {
        location: 'query',
        param: '',
        msg: 'Limit must be between 0 and 50. Default is 25.',
        value: '',
      };
      const { status, message } = Errors.BadRequest;
      const error = new ErrorREST(status, message, errorData);
      throw error;
    }
  } catch (err) {
    throw err;
  }
};
export default checkLimit;
