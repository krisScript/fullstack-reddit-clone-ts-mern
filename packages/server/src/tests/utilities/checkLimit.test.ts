import checkLimit from '../../utilities/checkLimit';
describe('checkLimit', (): void => {
  it("shouldn't throw an error", (): void => {
    expect((): void => checkLimit(1)).not.toThrow();
    expect((): void => checkLimit(50)).not.toThrow();
    expect((): void => checkLimit(25)).not.toThrow();
  });
  it(`should throw an error`, (): void => {
    expect((): void => checkLimit(51)).toThrowErrorMatchingSnapshot();
    expect((): void => checkLimit(0)).toThrowErrorMatchingSnapshot();
  });
});
