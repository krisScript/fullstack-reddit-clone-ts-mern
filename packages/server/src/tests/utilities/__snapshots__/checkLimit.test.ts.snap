// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkLimit should throw an error 1`] = `"Request has wrong format."`;

exports[`checkLimit should throw an error 2`] = `"Request has wrong format."`;
